scripts/bin/update-packages

```diff
@@ -103,11 +103,92 @@ _update() {
 	termux_pkg_auto_update
 }
 
+declare -A _LATEST_TAGS=()
 declare -A _FAILED_UPDATES=()
 declare -a _ALREADY_SEEN=() # Array of packages successfully updated or skipped.
 
+: '
+ _fetch_and_cache_tags fetches all possible tags using termux_pkg_auto_update, but using Ninja build system. 
+ The key difference is that we make the process concurrent, allowing us to fetch tags simultaneously rather than one at a time. 
+ Once all tags are cached, the termux_pkg_auto_update function will operate much more quickly. 
+ We avoid packages with overwritten termux_pkg_auto_update to prevent unexpected modifications to the package`s build.sh.
+'
+_fetch_and_cache_tags() {
+	echo "INFO: Fetching and caching tags"
+
+	# First invocation of termux_repology_api_get_latest_version fetches and caches repology metadata.
+	&>- termux_repology_api_get_latest_version ' '
```

Comment:
> I think we're redirecting stdout to stderr?

No, `&>-` closes both `stdout` and `stderr` since we do not really need an answer of `termux_repology_api_get_latest_version`, it will complain about non-existent package or simply report `null`. This call should be done only to create `$TERMUX_REPOLOGY_DATA_FILE` file.

Reply:
> No, &>- closes both `stdout` and `stderr`

Ah right.
Forgot about that.

Might be good to clarify that with a comment, at least once.